include/seqan3/alignment/configuration/align_config_method.hpp

```diff
@@ -28,12 +29,60 @@ namespace seqan3::align_cfg
  *
  * \details
  *
- * The alignment algorithm can be categorised in different methods. For example, the
- * \ref seqan3::align_cfg::method_local "local" and the
- * \ref seqan3::align_cfg::method_global "global" alignment are two different methods, while the semi-global alignment
- * is a variation of the global alignment. This differentiation makes it possible to define a subset of configurations
- * that can work with a particular method. Since it is not possible to guess what the desired method for a user is,
- * there is no default and this configuration must always be provided for the alignment algorithm.
+ * There are several methods for sequence alignment. We distinguish between \ref seqan3::align_cfg::method_local "local"
+ * and \ref seqan3::align_cfg::method_global "global" alignments. The semi-global alignment is implemented as a
+ * variation of the global alignment.
+ *
+ * **Global Alignment**:\verbatim
+--T--CC-C-AGT--TATGT-CAGGGGACACG-A-GCATGCAGA-GAC
+  |  || |  ||  | | | |||    || | | |  | ||||   |
+AATTGCCGCC-GTCGT-T-TTCAG----CA-GTTATG-T-CAGAT--C\endverbatim
+ * Finding the optimal global alignment of two sequences is solved by the **Needleman-Wunsch algorithm**.
+ *
+ * **Local Alignment** (better suited to find conserved segment):
+ * \verbatim
+                  tccCAGTTATGTCAGgggacacgagcatgcagagac
+                     ||||||||||||
+aattgccgccgtcgttttcagCAGTTATGTCAGatc
+\endverbatim
+ * A \ref seqan3::align_cfg::method_local "local" alignment is effectively a global alignment of two partial sequences,
+ * that is, when two genes from different species are similar in short conserved regions and dissimilar in the remaining
+ * regions. A global alignment would not find the local matching because it would try to align the entire sequence.
+ * This is solved by the **Smith-Waterman algorithm**.
+ *
+ * **Semi-global Alignment** (e.g. overlapping sequences):
+ * \verbatim
+                  TCCCAGTTATGTCAGgggacacgagcatgcagagac
+                  |||||||||||||||
+aattgccgccgtcgttttTCCCAGTTATGTCAG
+\endverbatim
+ * The semi-global alignment is a specially configured global alignment.
+ *
+ * **Complexity** of both algorithms:
+ *
+ * We need to store (n+1)(m+1) numbers, with a constant number of calculations required for each number (three sums and
+ * one maximum). Therefore, the algorithm requires **O(nm)** time and memory.
+ *
+ * To reduce the time complexity you can use a \ref seqan3::align_cfg::band_fixed_size "banded" alignment.
```

Comment:
```suggestion
 * To reduce the time complexity you can use a \ref seqan3::align_cfg::band_fixed_size "banded" alignment. It reduces
 * the runtime by a constant although remaining quadratic and limiting the possible solutions slightly.
 *
 * You can speed up the computation significantly if you parallelize (todo reference option parallel) and simdify
 * (reference simd option) your alignment.
```
Is simd possible yet, via public configuration? @marehr

Reply:
We have a vectorise config and you need to add some compiler options to enable machine optimisation like `-march=native` or `-mavx2`.

Reply:
should I link to seqan3::detail::simd_affine_gap_policy or what do you mean?

Reply:
What would be the right link? I get a warning with the one above..
```bash
warning: unable to resolve reference to 'seqan3::simd' for \ref command
```

Reply:
seqan3::simd is dev-only documented.